### ISSUE TITLE
Operator deployed Prometheus pod is restarting the whole time

### ISSUE THREAD

**Issue by 21stio** [coreos/prometheus-operator#381]
**What did you do?**

Deployed operator with coreos/prometheus-operator/bundle.yaml. Created prometheus object

**What did you expect to see?**

A healthy prometheus pod

**What did you see instead? Under which circumstances?**

The `prometheus-prometheus-0`'s containers are restarting (terminating, containercreating, running, termi..) the whole time. I deleted the Operator what made the `prometheus-prometheus-0` to stay alive.

* Environment: 
gke v 1.6.2

* Kubernetes version information:
```
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.3", GitCommit:"0480917b552be33e2dba47386e51decb1a211df6", GitTreeState:"clean", BuildDate:"2017-05-10T23:29:08Z", GoVersion:"go1.8.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:22:08Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

	gke

* Manifests:

` coreos/prometheus-operator/bundle.yaml`

Prometheus object

```yaml
apiVersion: monitoring.coreos.com/v1alpha1
kind: Prometheus
metadata:
  name: prometheus
spec:
  serviceMonitorSelector:
    matchLabels:
      team: frontend
  version: v1.6.3
  resources:
    requests:
      memory: 400Mi
```

* Prometheus Operator Logs:

```
ts=2017-05-21T08:09:41Z caller=operator.go:965 component=prometheusoperator msg="updating config skipped, no configuration change"
ts=2017-05-21T08:09:41Z caller=operator.go:579 component=prometheusoperator msg="update handler" old=245184 cur=245185
ts=2017-05-21T08:09:41Z caller=operator.go:316 component=alertmanageroperator msg="update handler" old=245184 cur=245185
ts=2017-05-21T08:09:41Z caller=operator.go:614 component=prometheusoperator msg="sync prometheus" key=default/prometheus
ts=2017-05-21T08:09:41Z caller=operator.go:965 component=prometheusoperator msg="updating config skipped, no configuration change"
ts=2017-05-21T08:09:41Z caller=operator.go:579 component=prometheusoperator msg="update handler" old=245185 cur=245186
ts=2017-05-21T08:09:41Z caller=operator.go:316 component=alertmanageroperator msg="update handler" old=245185 cur=245186
ts=2017-05-21T08:09:41Z caller=operator.go:614 component=prometheusoperator msg="sync prometheus" key=default/prometheus
ts=2017-05-21T08:09:41Z caller=operator.go:965 component=prometheusoperator msg="updating config skipped, no configuration change"
ts=2017-05-21T08:09:41Z caller=operator.go:579 component=prometheusoperator msg="update handler" old=245186 cur=245187
ts=2017-05-21T08:09:41Z caller=operator.go:316 component=alertmanageroperator msg="update handler" old=245186 cur=245187
ts=2017-05-21T08:09:41Z caller=operator.go:614 component=prometheusoperator msg="sync prometheus" key=default/prometheus
ts=2017-05-21T08:09:41Z caller=operator.go:965 component=prometheusoperator msg="updating config skipped, no configuration change"
```

`kubectl logs -c prometheus prometheus-prometheus-0`

```
time="2017-05-21T08:10:28Z" level=info msg="Starting prometheus (version=1.6.3, branch=master, revision=c580b60c67f2c5f6b638c3322161bcdf6d68d7fc)" source="main.go:88"
time="2017-05-21T08:10:28Z" level=info msg="Build context (go=go1.8.1, user=root@93eaf9195be6, date=20170519-13:03:54)" source="main.go:89"
time="2017-05-21T08:10:28Z" level=info msg="Loading configuration file /etc/prometheus/config/prometheus.yaml" source="main.go:251"
time="2017-05-21T08:10:28Z" level=info msg="Loading series map and head chunks..." source="storage.go:421"
time="2017-05-21T08:10:28Z" level=info msg="0 series loaded." source="storage.go:432"
time="2017-05-21T08:10:28Z" level=info msg="Listening on :9090" source="web.go:259"
time="2017-05-21T08:10:28Z" level=info msg="Starting target manager..." source="targetmanager.go:61"
time="2017-05-21T08:10:28Z" level=info msg="Loading configuration file /etc/prometheus/config/prometheus.yaml" source="main.go:251"
time="2017-05-21T08:10:31Z" level=warning msg="Received SIGTERM, exiting gracefully..." source="main.go:233"
time="2017-05-21T08:10:31Z" level=info msg="See you next time!" source="main.go:240"
time="2017-05-21T08:10:31Z" level=info msg="Stopping target manager..." source="targetmanager.go:75"
time="2017-05-21T08:10:31Z" level=info msg="Stopping rule manager..." source="manager.go:388"
time="2017-05-21T08:10:31Z" level=info msg="Rule manager stopped." source="manager.go:394"
time="2017-05-21T08:10:31Z" level=info msg="Stopping notification handler..." source="notifier.go:409"
time="2017-05-21T08:10:31Z" level=info msg="Stopping local storage..." source="storage.go:450"
time="2017-05-21T08:10:31Z" level=info msg="Stopping maintenance loop..." source="storage.go:452"
time="2017-05-21T08:10:31Z" level=info msg="Maintenance loop stopped." source="storage.go:1439"
time="2017-05-21T08:10:31Z" level=info msg="Stopping series quarantining..." source="storage.go:456"
time="2017-05-21T08:10:31Z" level=info msg="Series quarantining stopped." source="storage.go:1888"
time="2017-05-21T08:10:31Z" level=info msg="Stopping chunk eviction..." source="storage.go:460"
time="2017-05-21T08:10:31Z" level=info msg="Chunk eviction stopped." source="storage.go:1134"
time="2017-05-21T08:10:31Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:633"
time="2017-05-21T08:10:31Z" level=info msg="Done checkpointing in-memory metrics and chunks in 3.21255ms." source="persistence.go:665"
time="2017-05-21T08:10:31Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1526"
time="2017-05-21T08:10:31Z" level=info msg="Done checkpointing fingerprint mappings in 4.957027ms." source="persistence.go:1549"
time="2017-05-21T08:10:31Z" level=info msg="Local storage stopped." source="storage.go:477"
```

`kubectl logs -c prometheus-config-reloader prometheus-prometheus-0`

```
ts=2017-05-21T08:11:28Z caller=main.go:227 component=volume-watcher msg=Starting...
ts=2017-05-21T08:11:28Z caller=main.go:202 component=volume-watcher msg="Updating rule files..."
ts=2017-05-21T08:11:28Z caller=main.go:207 component=volume-watcher msg="Rule files updated."
ts=2017-05-21T08:11:28Z caller=main.go:210 component=volume-watcher msg="Reloading Prometheus..."
ts=2017-05-21T08:11:28Z caller=main.go:215 component=volume-watcher msg="Prometheus successfully reloaded."
```

**Comment by galexrt** [comment]
I'm experiencing the same issue with Kubernetes `v1.5.3` and Operator version `quay.io/coreos/prometheus-operator:v0.9.1`. I have a replica of 2 and the first one is always killed and restarted after ranging from 2 to 4 minutes.

**Comment by fabxc** [comment]
Which operator version did this start with for you?

**Comment by brancz** [comment]
hmm .. I cannot reproduce .. @21stio can you give us the output of `kubectl get pod prometheus-prometheus-0 -oyaml` .. maybe the status object can tell us why the container/pod was restarted

**Comment by 21stio** [comment]
@fabxc I'm this image `ec264f462e568b53de2519a062c3644d
    Image:`

@brancz the status object is changing, frequently 

```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    kubernetes.io/created-by: |
      {"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"StatefulSet","namespace":"default","name":"prometheus-prometheus","uid":"d2ef10c8-3df4-11e7-9057-42010a800061","apiVersion":"apps","resourceVersion":"788690"}}
    kubernetes.io/limit-ranger: 'LimitRanger plugin set: cpu request for container
      prometheus'
    pod.beta.kubernetes.io/hostname: prometheus-prometheus-0
    pod.beta.kubernetes.io/subdomain: prometheus-operated
  creationTimestamp: 2017-05-22T11:45:47Z
  deletionGracePeriodSeconds: 600
  deletionTimestamp: 2017-05-22T11:55:49Z
  generateName: prometheus-prometheus-
  labels:
    app: prometheus
    prometheus: prometheus
  name: prometheus-prometheus-0
  namespace: default
  ownerReferences:
  - apiVersion: apps/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: StatefulSet
    name: prometheus-prometheus
    uid: d2ef10c8-3df4-11e7-9057-42010a800061
  resourceVersion: "788723"
  selfLink: /api/v1/namespaces/default/pods/prometheus-prometheus-0
  uid: 32448d2b-3ee4-11e7-9057-42010a800061
spec:
  containers:
  - args:
    - -storage.local.retention=24h
    - -storage.local.num-fingerprint-mutexes=4096
    - -storage.local.path=/var/prometheus/data
    - -storage.local.chunk-encoding-version=2
    - -config.file=/etc/prometheus/config/prometheus.yaml
    - -storage.local.target-heap-size=279620266
    image: quay.io/prometheus/prometheus:v1.6.3
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 10
      httpGet:
        path: /status
        port: web
        scheme: HTTP
      initialDelaySeconds: 300
      periodSeconds: 5
      successThreshold: 1
      timeoutSeconds: 3
    name: prometheus
    ports:
    - containerPort: 9090
      name: web
      protocol: TCP
    readinessProbe:
      failureThreshold: 6
      httpGet:
        path: /status
        port: web
        scheme: HTTP
      periodSeconds: 5
      successThreshold: 1
      timeoutSeconds: 3
    resources:
      requests:
        cpu: 100m
        memory: 400Mi
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /etc/prometheus/config
      name: config
      readOnly: true
    - mountPath: /etc/prometheus/rules
      name: rules
      readOnly: true
    - mountPath: /var/prometheus/data
      name: prometheus-prometheus-db
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-t2zc4
      readOnly: true
  - args:
    - -reload-url=http://localhost:9090/-/reload
    - -config-volume-dir=/etc/prometheus/config
    - -rule-volume-dir=/etc/prometheus/rules
    image: quay.io/coreos/prometheus-config-reloader:v0.0.1
    imagePullPolicy: IfNotPresent
    name: prometheus-config-reloader
    resources:
      limits:
        cpu: 5m
        memory: 10Mi
      requests:
        cpu: 5m
        memory: 10Mi
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /etc/prometheus/config
      name: config
      readOnly: true
    - mountPath: /etc/prometheus/rules
      name: rules
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-t2zc4
      readOnly: true
  dnsPolicy: ClusterFirst
  nodeName: gke-cluster-1-pool-1-5957ca27-5nwd
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 600
  tolerations:
  - effect: NoExecute
    key: node.alpha.kubernetes.io/notReady
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.alpha.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - name: config
    secret:
      defaultMode: 420
      secretName: prometheus-prometheus
  - emptyDir: {}
    name: rules
  - emptyDir: {}
    name: prometheus-prometheus-db
  - name: default-token-t2zc4
    secret:
      defaultMode: 420
      secretName: default-token-t2zc4
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: 2017-05-22T11:45:47Z
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: 2017-05-22T11:45:49Z
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: 2017-05-22T11:45:47Z
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://945e9a9b23254c5d5b708637a43a3d03735129a7cec9712e2105fa3c839df147
    image: quay.io/prometheus/prometheus:v1.6.3
    imageID: docker://sha256:bc6b20b76bf998c6bb186949ef10d1bc85927e180f52a6e203438988fe051547
    lastState: {}
    name: prometheus
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2017-05-22T11:45:48Z
  - containerID: docker://d3dc8525cecdd93edd81683ffb28f12614c4b70727dce0437cac904f5ca1ba21
    image: quay.io/coreos/prometheus-config-reloader:v0.0.1
    imageID: docker://sha256:50533ddf053d92f17f864be7ea8d30e41b62229ae75b1a0fc3f6c3dcb5a9957b
    lastState: {}
    name: prometheus-config-reloader
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2017-05-22T11:45:48Z
  hostIP: 10.128.0.3
  phase: Running
  podIP: 10.0.1.242
  qosClass: Burstable
  startTime: 2017-05-22T11:45:47Z
```

```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    kubernetes.io/created-by: |
      {"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"StatefulSet","namespace":"default","name":"prometheus-prometheus","uid":"d2ef10c8-3df4-11e7-9057-42010a800061","apiVersion":"apps","resourceVersion":"789556"}}
    kubernetes.io/limit-ranger: 'LimitRanger plugin set: cpu request for container
      prometheus'
    pod.beta.kubernetes.io/hostname: prometheus-prometheus-0
    pod.beta.kubernetes.io/subdomain: prometheus-operated
  creationTimestamp: 2017-05-22T11:48:22Z
  deletionGracePeriodSeconds: 600
  deletionTimestamp: 2017-05-22T11:58:27Z
  generateName: prometheus-prometheus-
  labels:
    app: prometheus
    prometheus: prometheus
  name: prometheus-prometheus-0
  namespace: default
  ownerReferences:
  - apiVersion: apps/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: StatefulSet
    name: prometheus-prometheus
    uid: d2ef10c8-3df4-11e7-9057-42010a800061
  resourceVersion: "789591"
  selfLink: /api/v1/namespaces/default/pods/prometheus-prometheus-0
  uid: 8e9b14e7-3ee4-11e7-9057-42010a800061
spec:
  containers:
  - args:
    - -storage.local.retention=24h
    - -storage.local.num-fingerprint-mutexes=4096
    - -storage.local.path=/var/prometheus/data
    - -storage.local.chunk-encoding-version=2
    - -config.file=/etc/prometheus/config/prometheus.yaml
    - -storage.local.target-heap-size=279620266
    image: quay.io/prometheus/prometheus:v1.6.3
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 10
      httpGet:
        path: /status
        port: web
        scheme: HTTP
      initialDelaySeconds: 300
      periodSeconds: 5
      successThreshold: 1
      timeoutSeconds: 3
    name: prometheus
    ports:
    - containerPort: 9090
      name: web
      protocol: TCP
    readinessProbe:
      failureThreshold: 6
      httpGet:
        path: /status
        port: web
        scheme: HTTP
      periodSeconds: 5
      successThreshold: 1
      timeoutSeconds: 3
    resources:
      requests:
        cpu: 100m
        memory: 400Mi
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /etc/prometheus/config
      name: config
      readOnly: true
    - mountPath: /etc/prometheus/rules
      name: rules
      readOnly: true
    - mountPath: /var/prometheus/data
      name: prometheus-prometheus-db
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-t2zc4
      readOnly: true
  - args:
    - -reload-url=http://localhost:9090/-/reload
    - -config-volume-dir=/etc/prometheus/config
    - -rule-volume-dir=/etc/prometheus/rules
    image: quay.io/coreos/prometheus-config-reloader:v0.0.1
    imagePullPolicy: IfNotPresent
    name: prometheus-config-reloader
    resources:
      limits:
        cpu: 5m
        memory: 10Mi
      requests:
        cpu: 5m
        memory: 10Mi
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /etc/prometheus/config
      name: config
      readOnly: true
    - mountPath: /etc/prometheus/rules
      name: rules
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-t2zc4
      readOnly: true
  dnsPolicy: ClusterFirst
  nodeName: gke-cluster-1-pool-1-5957ca27-5nwd
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 600
  tolerations:
  - effect: NoExecute
    key: node.alpha.kubernetes.io/notReady
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.alpha.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - name: config
    secret:
      defaultMode: 420
      secretName: prometheus-prometheus
  - emptyDir: {}
    name: rules
  - emptyDir: {}
    name: prometheus-prometheus-db
  - name: default-token-t2zc4
    secret:
      defaultMode: 420
      secretName: default-token-t2zc4
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: 2017-05-22T11:48:22Z
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: 2017-05-22T11:48:28Z
    message: 'containers with unready status: [prometheus prometheus-config-reloader]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: 2017-05-22T11:48:22Z
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://84e9b301bac151097b263d1d0328e8b8d23e76d1d4dc260ca45107ae4b7f6208
    image: quay.io/prometheus/prometheus:v1.6.3
    imageID: docker://sha256:bc6b20b76bf998c6bb186949ef10d1bc85927e180f52a6e203438988fe051547
    lastState: {}
    name: prometheus
    ready: false
    restartCount: 0
    state:
      terminated:
        containerID: docker://84e9b301bac151097b263d1d0328e8b8d23e76d1d4dc260ca45107ae4b7f6208
        exitCode: 0
        finishedAt: 2017-05-22T11:48:27Z
        reason: Completed
        startedAt: null
  - containerID: docker://4a16764e09eadc7e71bc1a8d2c4e436b7df46adf12b0e63536fc2ac333a31003
    image: quay.io/coreos/prometheus-config-reloader:v0.0.1
    imageID: docker://sha256:50533ddf053d92f17f864be7ea8d30e41b62229ae75b1a0fc3f6c3dcb5a9957b
    lastState: {}
    name: prometheus-config-reloader
    ready: false
    restartCount: 0
    state:
      terminated:
        containerID: docker://4a16764e09eadc7e71bc1a8d2c4e436b7df46adf12b0e63536fc2ac333a31003
        exitCode: 2
        finishedAt: 2017-05-22T11:48:27Z
        reason: Error
        startedAt: null
  hostIP: 10.128.0.3
  phase: Running
  qosClass: Burstable
  startTime: 2017-05-22T11:48:22Z
```

**Comment by galexrt** [comment]
For me this also started with `0.9.1`. As mentioned when the Operator isn't running after creating a Prometheus instance it works fine (= the container isn't killed).

**Comment by brancz** [comment]
@galexrt are you also on a GKE cluster?

**Comment by galexrt** [comment]
@brancz I'm on a bare metal Kubernetes cluster (2 masters, 4 nodes; deployed with kubernetes/contrib Ansible; Kubernetes version: 1.5.3).

**Comment by brancz** [comment]
This suggests that it has to do with the config reload container crashing:

```
  - containerID: docker://4a16764e09eadc7e71bc1a8d2c4e436b7df46adf12b0e63536fc2ac333a31003
    image: quay.io/coreos/prometheus-config-reloader:v0.0.1
    imageID: docker://sha256:50533ddf053d92f17f864be7ea8d30e41b62229ae75b1a0fc3f6c3dcb5a9957b
    lastState: {}
    name: prometheus-config-reloader
    ready: false
    restartCount: 0
    state:
      terminated:
        containerID: docker://4a16764e09eadc7e71bc1a8d2c4e436b7df46adf12b0e63536fc2ac333a31003
        exitCode: 2
        finishedAt: 2017-05-22T11:48:27Z
        reason: Error
        startedAt: null
```

Also I'm unsure where these tolerations are coming from: 

```
  tolerations:
  - effect: NoExecute
    key: node.alpha.kubernetes.io/notReady
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.alpha.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
```

Although I don't think this has anything to do with it maybe you can try increasing the memory request? Depending on your cluster size 400Mi might be a little too low.

**Comment by Karnich** [comment]
I ran into this problem just now. Where prometheus-k8s-0 was constantly restarting duo to receiving SIGTERM. I traced it down to me having an old prometheus instance running in another namespace. When i removed that one the restarting stopped.

**Comment by brancz** [comment]
Interesting @Karnich ! Sounds like a deployment did not go through correctly. Can you describe in more detail? Was this a Pod that was left over? A StatefulSet? Was it in terminated state, etc.?

**Comment by 21stio** [comment]
@brancz just tried w/ `requests.memory: 1000Mi` on the prometheus object, same behaviour.. 

I can give you access to my cluster, that one is just a playground

**Comment by Karnich** [comment]
@brancz it was a leftover deployment i believe.. I actually do not remember the details.. I just remembered seeing a prometheus pod in a namespace it did not belong to. And i think it was a deployment and replicaset but im not 100%. It was just guess work to remove it and see if it was the reason to the restarting.

It was living well with prometheus-operator 0.8 but was interfering with 0.9. weird part was that if i wanted 2 replicas.. the one with number 1 would come up but 0 would restart indefinitely

**Comment by 21stio** [comment]
Also just deployed v0.8, it's working fine now..

**Comment by brancz** [comment]
Fascinating, I have multiple clusters running 0.9.1 and have not been able to reproduce. I am currently busy otherwise, does your offer of debugging on your cluster still stand @21stio ?

**Comment by codesuki** [comment]
Same thing happens here too. I tried starting with a fresh cluster and applied the configs again. Same. Weird that as @Karnich said above, pod-0 gets killed but pod-1 runs fine.

**Comment by codesuki** [comment]
After some investigating I found this pullreq https://github.com/coreos/prometheus-operator/pull/331
Seems to be related. After checking what it does (comparing spec), I compared the spec myself doing `kubectl describe statefulset prometheus-main` and `kubectl describe pod prometheus-main-0`.
I noticed that the pod had a default value of 
```    
Requests:
      cpu:		100m
```

Which the `spec` was missing because my spec looks like this: 
```
  resources:
    requests:
      memory: 400Mi
```

After adding cpu like

```
  resources:
    requests:
      cpu: 100m
      memory: 400Mi
```
The pod is not getting killed anymore.
So, the spec/pod diff might be different for you guys but to find out what is missing just describe both.

The algorithm that was introduced in the pullreq should probably be refined to respect defaults or something like that.

**Comment by brancz** [comment]
Great find @codesuki! That makes a whole lot of sense, I'll make sure we fix and release ASAP.

**Comment by codesuki** [comment]
I love how the changelog says 
```
[ENHANCEMENT] Improved change detection to trigger Prometheus rollout.
```
The irony... hahaha

**Comment by fabxc** [comment]
Well, it was meant to trigger it more often, looks like we went beyond success with that :P
But seriously, thanks for debugging – fix will follow.

**Comment by codesuki** [comment]
Hahaha yeah, I am not criticizing. Actually had a good laugh 👍

**Comment by brancz** [comment]
none taken - it is indeed ironic :smile:

**Comment by 21stio** [comment]
great :)

**Comment by fabxc** [comment]
As @brancz pointed out, rolling update support for StatefulSet is planned work upstream (https://github.com/kubernetes/community/pull/503/files), which does basically what we would implement here – just that we have to work around a bunch of complications for that.

I'd propose rolling back the resource part of #331 for now and wait for upstream handling of this. This just means that resource adjustments need to be handled by users for the time being, by sequentially deleting the Prometheus pods, so they get restarted with the new resource requests.

**Comment by brancz** [comment]
This was fixed by https://github.com/coreos/prometheus-operator/pull/385 and has been released in `v0.10.0`.
